central/networkgraph/flow/datastore/internal/store/postgres/store.go

```diff
@@ -499,14 +510,24 @@ func (s *flowStoreImpl) GetFlowsForDeployment(ctx context.Context, deploymentID
 	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.GetFlowsForDeployment, "NetworkFlow")
 
 	return pgutils.Retry2(ctx, func() ([]*storage.NetworkFlow, error) {
-		return s.retryableGetFlowsForDeployment(ctx, deploymentID)
+		return s.retryableGetFlowsForDeployment(ctx, deploymentID, getByDeploymentStmt)
 	})
 }
-func (s *flowStoreImpl) retryableGetFlowsForDeployment(ctx context.Context, deploymentID string) ([]*storage.NetworkFlow, error) {
+
+// GetExternalFlowsForDeployment returns the external flows matching the deployment ID
+func (s *flowStoreImpl) GetExternalFlowsForDeployment(ctx context.Context, deploymentID string) ([]*storage.NetworkFlow, error) {
+	defer metrics.SetPostgresOperationDurationTime(time.Now(), ops.GetFlowsForDeployment, "NetworkFlow")
```

Comment:
Should a new operation metric be needed for that function (Get**External**FlowsForDeployment)

Reply:
IMO, a good point.  Though not super important so I would write a tech debt ticket and do it as follow up work eventually.

Reply:
I have created a new metric as suggested at https://github.com/stackrox/stackrox/pull/12689/commits/ead68617ccaee0fa541dfcd467d52cd13786ba31